DigitalPassaport/DigitalPassaport/Infactuture/Networking/NetworkingService.swift

```diff
@@ -11,19 +11,37 @@ import Foundation
 
 protocol NetworkingServiceProtocol {
     func publisherForRequest<Request, Response>(
-        router: NetworkManager.Router,
+        router: Router,
         request: Request?
     ) -> AnyPublisher<Response, APIError> where Request: Encodable, Response: Decodable
 }
 
 class NetworkingService: NetworkingServiceProtocol {
-    private let networkManager = NetworkManager()
-
+    private let session: Session
+    
+    init() {
+        let configuration = URLSessionConfiguration.default
+        configuration.timeoutIntervalForRequest = 30
+        self.session = Session(configuration: configuration)
+    }
+    
     func publisherForRequest<Request, Response>(
-        router: NetworkManager.Router,
+        router: Router,
```

Comment:
If you made it generic, than parameter should be the type of protocol NetworkRoutable, in that way it will be satisfied with any Network Router that implement that protocol